aspnetcore/fundamentals/minimal-apis/bindStreamPipeReader/7.0-samples/PipeStreamToBackgroundQueue/Program.cs

```diff
@@ -0,0 +1,31 @@
+using System.Threading.Channels;
+using BackgroundQueueService;
+var builder = WebApplication.CreateBuilder(args);
+// Create a channel to send data to the background queue.
+builder.Services.AddSingleton<Channel<Memory<byte>>>((_) => Channel.CreateBounded<Memory<byte>>(100));
+
+// Create a background queue service.
+builder.Services.AddHostedService<BackgroundQueue>();
+var app = builder.Build();
+
+
+// curl --request POST 'http://localhost:5256/register' --header 'Content-Type: application/json' --data-raw '{ "Name":"Samson", "Age": 23, "Country":"Nigeria" }'
+app.MapPost("/register", async (Stream body, Channel<Memory<byte>> queue) =>
+{
+    // Create a rewindable stream to be able to reuse the body stream.
+    var reusableStream = new MemoryStream();
+
+    // Copy the request body to the reusable stream.
+    await body.CopyToAsync(reusableStream);
+
+    // Reset the stream to the beginning.
+    reusableStream.Position = 0;
+
+    // Send the stream to the background queue.
+    await queue.Writer.WriteAsync(reusableStream.ToArray());
```

Comment:
Now the magic to avoid double allocations:

```suggestion
        reusableStream.TryGetBuffer(out var buffer);
        await queue.Writer.WriteAsync(buffer);
```

Reply:
Clean Code 🥂